fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -489,8 +604,16 @@ def initialize(user = nil, password = nil, use_shell_script = false, provider_sh
       end
 
       @jwt = jwt
+      @api_key = api_key
+
+      if should_use_altool?(upload, use_shell_script)
+        UI.verbose("Using altool as transporter.")
+        @transporter_executor = AltoolTransporterExecutor.new
+      else
+        UI.verbose("Using iTMSTransporter as transporter.")
+        @transporter_executor = use_shell_script ? ShellScriptTransporterExecutor.new : JavaTransporterExecutor.new
+      end
```

Comment:
i just notice that `itunes_transporter_spec.rb` is failing in Xcode16. our Fastlane CircleCI is still using Xcode13 to run the unit-tests and thats why failed to report about it.

Adding steps to reproduce it: 
- Step1: Select Xcode16 `sudo xcode-select -s /Applications/Xcode16.app`
- Step2: Run unit-test `bundle exec rspec fastlane/fastlane_core/spec/itunes_transporter_spec.rb`


i am working on fixing these unit-tests to speed-up and to make this PR ready to merge in master branch asap.
Thank you again for all the hard-work! @freddi-kit ❤️

Reply:
- Fixed failing `itunes_transporter_spec.rb` unit tests https://github.com/fastlane/fastlane/pull/20631/commits/d36bf23402b6d4c6b4681fc73f355a7b6819f95e
- Fixed failing `helper_spec.rb` unit tests https://github.com/fastlane/fastlane/pull/20631/commits/e4e44ac18ed8a8b959d7383d157dd2e065087c02

Reply:
Thank you so much! 🙇